jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -803,28 +806,29 @@ public Action process() throws Exception
                             gatherWrite += 2;
                             bytes += _chunk.remaining();
                         }
-                        if (BufferUtil.hasContent(_content))
+                        long contentLength = getRemainingContent();
+                        if (contentLength > 0)
                         {
                             gatherWrite += 1;
-                            bytes += _content.remaining();
+                            bytes += contentLength;
                         }
                         HttpConnection.this.bytesOut.add(bytes);
                         switch (gatherWrite)
                         {
                             case 7:
-                                getEndPoint().write(this, _header, _chunk, _content);
+                                getEndPoint().write(this, ArrayUtil.add(new ByteBuffer[]{_header, _chunk}, _content));
```

Comment:
This could probably be more efficient, as you are allocating a small array just to be added to the content.
Perhaps a special purpose prependToArray that takes 2 items to be prepended?